### PR TITLE
Pipeline model type check

### PR DESCRIPTION
**Pull request by JetRunner**
Add model type check for pipelines.

https://github.com/huggingface/transformers/issues/5678

### CODE REVIEW

**Review by sgugger — APPROVED** [review]

File: src/transformers/pipelines.py
Comment (sshleifer) [inline src/transformers/pipelines.py:619]: why is this change needed?

File: src/transformers/pipelines.py
Comment (sshleifer) [inline src/transformers/pipelines.py:489]: I would warn rather than raise an error, because raising an exception here requires us to keep `supported_models` completely up to date.

File: src/transformers/pipelines.py
Comment (JetRunner) [reply src/transformers/pipelines.py:619]: No it's not. I've rolled it back.

File: src/transformers/pipelines.py
Comment (JetRunner) [reply src/transformers/pipelines.py:489]: For all pipelines except `TextGenerationPipeline`, `supported_models` are automatically constructed from `AutoModel` which is certainly up-to-date. Plus, `pipeline` is a "production" style API, using a newly added model seems to be a corner case.

Comment (JetRunner) [reply src/transformers/pipelines.py:489]: Also, I throw errors following the original logic of `SummarizationPipeline`, which maintains an `ALLOW_LIST` and throws an error if the model is not on that list.

**Review by sshleifer — APPROVED** [review]
LGTM

**Comment by julien-c** [comment]
This is a reasonable change, however next time on Pipelines please wait for review from @mfuntowicz, @LysandreJik or I before merging

(especially as it can impact the inference API)

**Comment by JetRunner** [comment]
> This is a reasonable change, however next time on Pipelines please wait for review from @mfuntowicz, @LysandreJik or I before merging
> 
> 
> 
> (especially as it can impact the inference API)

Ok, I wasn't aware of that.

**Pull request merged**
